### PR TITLE
Added Firebase Authentication

### PR DESCRIPTION
**Pull request by ctstewart**
Added very basic authentication with Firebase. As of right now, the app redirects the user to Firebase's default login page if they're not logged in. This page currently allows users to login or register to the app. Once they're logged in, it continues the app's flow to the Story component.

### CODE REVIEW

**Review by afogel — COMMENTED** [review]
overall code looks good, just remove the LOGIN_wCHAR png. Also, in lieu of automated tests, I would like to see an entry for a regression test.
Here is the template for the regression test:
https://github.com/UW-MAGIC-lab/hidden_village/wiki/Regression-Tests#links-to-different-regression-tests

File: package.json [hunk 1]
```diff
@@ -24,9 +24,11 @@
     "eslint-plugin-jsx-a11y": "^6.4.1",
     "eslint-plugin-react": "^7.26.0",
     "jsdoc": "^3.6.7",
-    "parcel": "^2.0.0-rc.0",
+    "parcel": "^2.8.3",
     "postcss": "^8.3.8",
     "prettier": "^2.4.1",
+    "process": "^0.11.10",
```
[the remark this answers is not in the document]
Comment (ctstewart) [reply package.json:30]: Which two? My end only shows one new one, "process." I think that was needed with FirebaseUI, but should no longer be needed. I'll double check, and if it's not, I'll update the repo.

**Comment by afogel** [comment]
Looks like there's also a merge conflict in `.postcssrc`. Please correct 🙏

File: src/firebase/database.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Firebase Init
+import { initializeApp } from "firebase/database";
```
Comment (afogel) [inline src/firebase/database.js:2]: unused? Do we need this file?

File: src/components/auth/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    signInWithEmailAndPassword(auth, email, password)
+    .then((userCredential) => {
+      const user = userCredential.user;
+      window.location.href = "/";
+      console.log("success!");
+    })
+    .catch((error) => {
+      // Todo: add UI change on wrong password,
+      // error code 400
+      const errorCode = error.code;
```
Comment (afogel) [inline src/components/auth/SignIn.js:34]: probably want to `console.error` both the code and the message, if you're going to log anything to the console.

File: src/components/auth/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value)
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    signInWithEmailAndPassword(auth, email, password)
+    .then((userCredential) => {
+      const user = userCredential.user;
```
Comment (afogel) [inline src/components/auth/SignIn.js:27]: what is this being used for?

File: src/components/auth/SignIn.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      window.location.href = "/";
+      console.log("success!");
```
Comment (afogel) [inline src/components/auth/SignIn.js:29]: typically, we try to remove `console.log`. remove in next go around?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (afogel) [reply package.json:30]: the second one is `punycode`

**Review by afogel — CHANGES_REQUESTED** [review]
Very close! Looks like we can remove the unused database file altogether, and since we're at it, clean up some of those other places in the code base, add a [regression test](https://github.com/UW-MAGIC-lab/hidden_village/wiki/Regression-Tests) and call it a day 😎

File: src/components/Story.js [hunk 6]
```diff
@@ -76,6 +76,7 @@ const Story = () => {
       minDetectionConfidence: 0.5,
       minTrackingConfidence: 0.5,
       selfieMode: true,
+      refineFaceLandmarks: true,
```
Comment (afogel) [inline src/components/Story.js:79]: interesting -- I didn't realize this was part of the API, will have to read into it!

File: src/components/auth/SignIn.js (hunk 5 above)
Comment (ctstewart) [reply src/components/auth/SignIn.js]: Done

File: src/components/auth/SignIn.js (hunk 4 above)
Comment (ctstewart) [reply src/components/auth/SignIn.js]: The user variable, or the whole chunk? The chunk is used to login the user to a Firebase session. The variable was used for debugging, it is now removed.

File: src/components/auth/SignIn.js (hunk 3 above)
Comment (ctstewart) [reply src/components/auth/SignIn.js]: Fixed, added the entire `error` object

File: src/firebase/database.js (hunk 2 above)
Comment (ctstewart) [reply src/firebase/database.js:2]: Yes, we're adding to it in other branches

**Pull request closed without merging**